masquerade-core/src/main/java/com/flipkart/masquerade/processor/RuleObjectProcessor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.masquerade.processor;
+
+import com.flipkart.masquerade.Configuration;
+import com.flipkart.masquerade.rule.Rule;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+
+import javax.lang.model.element.Modifier;
+import java.util.Collection;
+import java.util.Map;
+
+import static com.flipkart.masquerade.util.Helper.getRuleInterface;
+import static com.flipkart.masquerade.util.Strings.*;
+
+/**
+ * Processor which adds the entry point method in the Entry class
+ * <p />
+ * Created by shrey.garg on 12/05/17.
+ */
+public class RuleObjectProcessor {
+    private final Configuration configuration;
+    private final TypeSpec.Builder cloakBuilder;
+
+    /**
+     * @param configuration Configuration for the current processing cycle
+     * @param cloakBuilder Entry class under construction for the cycle
+     */
+    public RuleObjectProcessor(Configuration configuration, TypeSpec.Builder cloakBuilder) {
+        this.configuration = configuration;
+        this.cloakBuilder = cloakBuilder;
+    }
+
+    /**
+     * @param rule Rule for which the entry point will be added
+     */
+    public void addEntry(Rule rule) {
+        MethodSpec.Builder objectMaskBuilder = MethodSpec.methodBuilder(ENTRY_METHOD);
+        objectMaskBuilder.addModifiers(Modifier.PUBLIC);
+        /* Suppress warnings as the generated code will never cause a CLassCastException */
+        objectMaskBuilder.addAnnotation(AnnotationSpec.builder(SuppressWarnings.class).addMember("value", "$S", "unchecked").build());
+        /* Adds an Object class parameter for which all the process at runtime will happen */
+        objectMaskBuilder.addParameter(Object.class, OBJECT_PARAMETER);
+        /* The second parameter refers to the Evaluator Object which will be used for comparisons */
+        objectMaskBuilder.addParameter(rule.getEvaluatorClass(), EVAL_PARAMETER);
+
+        /* If a null Object is passed, return immediately */
+        objectMaskBuilder.beginControlFlow("if ($L == null)", OBJECT_PARAMETER);
```

Comment:
Have null check for EVAL_PARAMETER as well as it might lead to NPE somewhere in the generated cloak code (if eval object is null)